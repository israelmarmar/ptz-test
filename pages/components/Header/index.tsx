import styles from "../../../styles/Header.module.css";
import pokeball from "../../../public/images/white-pokeball.svg";
import { RedButton } from "../RedButton";
import { WhiteButton } from "../WhiteButton";

export default function Header() {
  return (
    <div className={styles.header}>
        <RedButton title="Centro pokÃ©mon" iconSrc={pokeball} isCollapse />
        <div className={styles["header-menu"]}>
            <WhiteButton title="Quem somos" link={"/about"} />
            <RedButton title="Agendar consulta" />
        </div>
    </div>
  );
}
