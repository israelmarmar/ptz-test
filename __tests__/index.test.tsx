import { render, screen } from '@testing-library/react'
import Header from '../components/Header'
import Home from '../pages'

describe('Home', () => {
  it('renders a heading', () => {
    render(<Home />)
    const heading = screen.getByText("Cuidamos bem do seu pokémon, para ele cuidar bem de você")
    expect(heading).toBeInTheDocument()
  })

  it('renders a background image', () => {
    render(<Home />)
    const main = screen.getByTestId("main-with-image")
    expect(main).toHaveStyle('background-image: url("/images/pokemon-hero.jpg") no-repeat')
  })

  it('renders logo', () => {
    render(<Header />)
    const component = screen.getByText("Centro pokémon")
    expect(component).toBeInTheDocument()
  })

  it('renders consulta button', () => {
    render(<Header />)
    const component = screen.getByText("Agendar consulta")
    expect(component).toBeInTheDocument()
  })

  it('renders about button', () => {
    render(<Header />)
    const component = screen.getByText("Quem somos")
    expect(component).toBeInTheDocument()
  })

})