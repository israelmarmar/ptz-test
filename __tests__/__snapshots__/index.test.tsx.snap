// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a heading 1`] = `
<div>
  <main
    class="main"
  >
    <h6>
      Cuidamos bem do seu pokémon, para ele cuidar bem de você
    </h6>
  </main>
</div>
`;
